src/main/io/serial_4way.c

```diff
@@ -428,487 +411,402 @@ void esc4wayProcess(serialPort_t *mspPort)
             ioMem.D_PTR_I = ParamBuf;
-
-
             switch (CMD) {
-                // ******* Interface related stuff *******
-                case cmd_InterfaceTestAlive:
-                {
-                    if (isMcuConnected()) {
-                        switch (CurrentInterfaceMode)
-                        {
-                            #ifdef USE_SERIAL_4WAY_BLHELI_BOOTLOADER
-                            case imATM_BLB:
-                            case imSIL_BLB:
-                            case imARM_BLB:
-                            {
-                                if (!BL_SendCMDKeepAlive()) { // SetStateDisconnected() included
-                                    ACK_OUT = ACK_D_GENERAL_ERROR;
-                                }
-                                break;
-                            }
-                            #endif
-                            #ifdef USE_SERIAL_4WAY_SK_BOOTLOADER
-                            case imSK:
-                            {
-                                if (!Stk_SignOn()) { // SetStateDisconnected();
-                                    ACK_OUT = ACK_D_GENERAL_ERROR;
-                                }
-                                break;
-                            }
-                            #endif
-                            default:
-                                ACK_OUT = ACK_D_GENERAL_ERROR;
+            // ******* Interface related stuff *******
+            case cmd_InterfaceTestAlive: {
+                if (isMcuConnected()) {
+                    switch (CurrentInterfaceMode) {
+#ifdef USE_SERIAL_4WAY_BLHELI_BOOTLOADER
+                    case imATM_BLB:
+                    case imSIL_BLB:
+                    case imARM_BLB: {
+                        if (!BL_SendCMDKeepAlive()) { // SetStateDisconnected() included
+                            ACK_OUT = ACK_D_GENERAL_ERROR;
                         }
-                        if ( ACK_OUT != ACK_OK) SET_DISCONNECTED;
+                        break;
+                    }
+#endif
+#ifdef USE_SERIAL_4WAY_SK_BOOTLOADER
+                    case imSK: {
+                        if (!Stk_SignOn()) { // SetStateDisconnected();
+                            ACK_OUT = ACK_D_GENERAL_ERROR;
+                        }
+                        break;
+                    }
+#endif
+                    default:
+                        ACK_OUT = ACK_D_GENERAL_ERROR;
+                    }
+                    if ( ACK_OUT != ACK_OK) SET_DISCONNECTED;
+                }
+                break;
+            }
+            case cmd_ProtocolGetVersion: {
+                // Only interface itself, no matter what Device
+                Dummy.bytes[0] = SERIAL_4WAY_PROTOCOL_VER;
+                break;
+            }
+            case cmd_InterfaceGetName: {
+                // Only interface itself, no matter what Device
+                // O_PARAM_LEN=16;
+                O_PARAM_LEN = strlen(SERIAL_4WAY_INTERFACE_NAME_STR);
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6326343)